src/client/InitialShape.cpp

```diff
@@ -28,7 +28,13 @@ InitialShape::InitialShape(const std::vector<double>& vert) : mVertices(vert), m
 }
 
 InitialShape::InitialShape(const std::vector<double>& vert, const std::vector<uint32_t>& ind,
-                           const std::vector<uint32_t>& faceCnt)
-    : mVertices(vert), mIndices(ind), mFaceCounts(faceCnt), mPathFlag(false) {}
+                           const std::vector<uint32_t>& faceCnt, const std::vector<std::vector<uint32_t>>& holes = {{}})
```

Comment:
consider introducing a type like HoleIndices for std::vector<std::vector<uint32_t>>